samples/snippets/src/main/java/com/example/contactcenterinsights/CreateAnalyzeConversation.java

```diff
@@ -0,0 +1,90 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.example.contactcenterinsights;
+
+// [START contactcenterinsights_create_analyze_conversation]
+import com.google.api.gax.longrunning.OperationFuture;
+import com.google.cloud.contactcenterinsights.v1.Analysis;
+import com.google.cloud.contactcenterinsights.v1.ContactCenterInsightsClient;
+import com.google.cloud.contactcenterinsights.v1.Conversation;
+import com.google.cloud.contactcenterinsights.v1.ConversationDataSource;
+import com.google.cloud.contactcenterinsights.v1.CreateAnalysisOperationMetadata;
+import com.google.cloud.contactcenterinsights.v1.CreateAnalysisRequest;
+import com.google.cloud.contactcenterinsights.v1.CreateConversationRequest;
+import com.google.cloud.contactcenterinsights.v1.DeleteConversationRequest;
+import com.google.cloud.contactcenterinsights.v1.GcsSource;
+import com.google.cloud.contactcenterinsights.v1.LocationName;
+
+public class CreateAnalyzeConversation {
+
+  public static void main(String[] args) {
+    String projectId = "your-project-id";
+    String location = "us-central1";
+    String conversationId = "your-conversation-id";
+    String transcriptUri = "gs://example-bucket/some-example.json";
+    Conversation.Medium medium = Conversation.Medium.CHAT;
+
+    // The conversation to create.
+    Conversation conversation =
+      Conversation.newBuilder()
+        .setDataSource(
+          ConversationDataSource.newBuilder()
+            .setGcsSource(GcsSource.newBuilder().setTranscriptUri(transcriptUri).build())
+            .build())
+        .setMedium(medium)
+        .build();
+
+    createAnalyzeConversation(projectId, location, conversationId, conversation);
+  }
+
+  public static void createAnalyzeConversation(
+    String projectId, String location, String conversationId, Conversation conversation) {
+    try (ContactCenterInsightsClient contactCenterInsightsClient =
+           ContactCenterInsightsClient.create()) {
+      LocationName parent = LocationName.of(projectId, location);
+
+      CreateConversationRequest.Builder createConversationRequestBuilder =
+        CreateConversationRequest.newBuilder()
+          .setConversation(conversation)
+          .setParent(parent.toString());
+      if (!conversationId.isEmpty()) {
+        createConversationRequestBuilder.setConversationId(conversationId);
+      }
+      Conversation response =
+        contactCenterInsightsClient.createConversation(createConversationRequestBuilder.build());
+
+      String conversationName = response.getName();
+      CreateAnalysisRequest request =
+        CreateAnalysisRequest.newBuilder().setParent(conversationName).build();
+      OperationFuture<Analysis, CreateAnalysisOperationMetadata> future =
+        contactCenterInsightsClient.createAnalysisAsync(request);
+
+      System.out.println("Waiting for operation to complete...");
+      Analysis analysis = future.get();
```

Comment:
Can you add a timeout here, so it doesn't block forever?

Reply:
Done.